src/logic/unique.lean

```diff
@@ -137,22 +137,35 @@ lemma unique_iff_subsingleton_and_nonempty (α : Sort u) :
 ⟨λ ⟨u⟩, by split; exactI infer_instance,
  λ ⟨hs, hn⟩, ⟨by { resetI, inhabit α, exact unique.mk' α }⟩⟩
 
-@[simp] lemma pi.default_def {β : Π a : α, Sort v} [Π a, inhabited (β a)] :
+@[simp] lemma pi.default_def {β : α → Sort v} [Π a, inhabited (β a)] :
   @default (Π a, β a) _ = λ a : α, @default (β a) _ := rfl
 
-lemma pi.default_apply {β : Π a : α, Sort v} [Π a, inhabited (β a)] (a : α) :
+lemma pi.default_apply {β : α → Sort v} [Π a, inhabited (β a)] (a : α) :
   @default (Π a, β a) _ a = default := rfl
 
-instance pi.unique {β : Π a : α, Sort v} [Π a, unique (β a)] : unique (Π a, β a) :=
+instance pi.unique {β : α → Sort v} [Π a, unique (β a)] : unique (Π a, β a) :=
 { uniq := λ f, funext $ λ x, unique.eq_default _,
   .. pi.inhabited α }
 
 /-- There is a unique function on an empty domain. -/
-instance pi.unique_of_is_empty [is_empty α] (β : Π a : α, Sort v) :
+instance pi.unique_of_is_empty [is_empty α] (β : α → Sort v) :
   unique (Π a, β a) :=
 { default := is_empty_elim,
   uniq := λ f, funext is_empty_elim }
 
+lemma congr_arg_heq_of_subsingleton [subsingleton α] {β : α → Sort v}
+  (f : Π a, β a) (i j : α) : f i == f j := by rw subsingleton.elim i j
+
+lemma congr_arg_of_subsingleton [subsingleton α]
+  (f : α → β) (i j : α) : f i = f j := by rw subsingleton.elim i j
+
+lemma eq_const_of_unique [unique α] (f : α → β) : f = λ _, f default :=
```

Comment:
Better or worse with `function.const` on the RHS?

Reply:
All uses of this lemma in the homotopy group PR are `exact`, so it doesn't matter, and I'll change it to `function.const` to conform better to the name.